osu.Game.Rulesets.Taiko/UI/TaikoPlayfield.cs

```diff
@@ -213,12 +222,28 @@ public override void Add(DrawableHitObject h)
                     barlineContainer.Add(barline.CreateProxy());
                     break;
 
+                case DrawableDrumRoll drumRoll:
+                    drumRoll.OnHit += onDrumrollArbitraryHit;
+                    break;
+
                 case DrawableTaikoHitObject taikoObject:
                     topLevelHitContainer.Add(taikoObject.CreateProxiedContent());
                     break;
             }
         }
 
+        private void onDrumrollArbitraryHit(TaikoAction action, bool isStrong)
+        {
+            DrawableHit drawableHit;
+
+            if (action == TaikoAction.LeftRim || action == TaikoAction.RightRim)
+                drawableHit = new DrawableFlyingRimHit(Time.Current, isStrong);
```

Comment:
After the previous requested changes have been applied, one improvement you can make here is to apply an offset of `-result.TimeOffset` here to make the flying hits spawn at the position of the ticks.

TimeOffset is calculated as `CurrentTime - HitObject.StartTime`, which is negative if the user hits before the start time. So the inverse offset will move the hits in the timeline just enough to appear on top of the ticks.